### ISSUE TITLE
Bug: ALZ-MGMT and ALZ-MGMT-TEMPLATES repositories are not populated (permissions)

### ISSUE THREAD

**Issue by DylanRobinson** [Azure/ALZ-PowerShell-Module#260]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Infrastructure as Code Type? (Required)

bicep

### PowerShell Module Version (Optional)

4.1.1

### Bootstrap Module Version (Optional)

4.1.4

### Starter Module? (Required)

bicep - complete

### Starter Module Version (Optional)

0.20.1

### Input arguments of the ALZ-PowerShell-Module (Optional)

 Deploy-Accelerator -inputs "e:\accelerator\config\inputs.yaml" -output "e:\accelerator\output"

### Debug Output/Panic Output (Optional)

```shell
╷
│ Error:  Creating repository in Azure DevOps: TF401019: The Git repository with name or identifier 29B5E1A8-F6AF-4E19-9186-BC542C6F7BA3 does not exist or you do not have p
ermissions for the operation you are attempting.
│
│   with module.azure_devops.azuredevops_git_repository.alz,
│   on ..\..\modules\azure_devops\repository_module.tf line 1, in resource "azuredevops_git_repository" "alz":
│    1: resource "azuredevops_git_repository" "alz" {
│
╵
╷
│ Error:  Creating repository in Azure DevOps: TF401019: The Git repository with name or identifier 1CFEB7D8-0F4B-4B95-8C43-AF8972E0FA2C does not exist or you do not have p
ermissions for the operation you are attempting.
│
│   with module.azure_devops.azuredevops_git_repository.alz_templates[0],
│   on ..\..\modules\azure_devops\repository_templates.tf line 1, in resource "azuredevops_git_repository" "alz_templates":
│    1: resource "azuredevops_git_repository" "alz_templates" {
│
╵

Terraform apply failed with exit code 1. This is likely a transient issue, so we are retrying...
```

then later

```shell
module.azure_devops.azuredevops_git_repository.alz_templates[0]: Creating...
module.azure_devops.azuredevops_git_repository.alz: Creating...
╷
│ Error:  Creating repository in Azure DevOps: TF400948: A Git repository with the name alz-mgmt already exists.
│
│   with module.azure_devops.azuredevops_git_repository.alz,
│   on ..\..\modules\azure_devops\repository_module.tf line 1, in resource "azuredevops_git_repository" "alz":
│    1: resource "azuredevops_git_repository" "alz" {
│
╵
╷
│ Error:  Creating repository in Azure DevOps: TF400948: A Git repository with the name alz-mgmt-templates already exists.
│
│   with module.azure_devops.azuredevops_git_repository.alz_templates[0],
│   on ..\..\modules\azure_devops\repository_templates.tf line 1, in resource "azuredevops_git_repository" "alz_templates":
│    1: resource "azuredevops_git_repository" "alz_templates" {
│
╵

Terraform apply failed with exit code 1. This is likely a transient issue, so we are retrying...
```

### Expected Behaviour (Required)

Successful deployment...

### Actual Behaviour (Required)

Azure DevOps project is created.
alz-mgmt and alz-mgmt-templates repositories are created but not populated / they are empty.
Error: Creating repository in Azure DevOps: TF400948: A Git repository with the name alz-mgmt already exists.
the same for alz-mgmt-templates.

### Steps to Reproduce (Optional)

Following the steps in the wiki [User Guide] Quick Start Azure DevOps

### Important Factoids (Optional)

Work being done with assistance from Microsoft CSAs as part of a 3-day Azure Landing Zone deployment accelerator.

**Comment by oZakari** [comment]
Hey @DylanRobinson, could you try re-running Deploy-Accelerator cmdlet over the top of the existing deployment as I believe that should resolve the transient issue you encountered.

**Comment by DylanRobinson** [comment]
Hi @oZakari - we've tried re-running several times over the deployment without success.

We've also tried deleting the repositories (they were re-created but again empty), deleting the DevOps project, destroying the environment (both manually and using the -destroy parameter, and clearing the output folder).

**Comment by DylanRobinson** [comment]
@oZakari I can confirm all permissions were set exactly as noted in that article, including Code: Full.

I have also tried giving my PAT Full Access just to rule out the possibility of any additional, documented, permission requirements.

**Comment by jaredfholgate** [comment]
I have not seen this issue. Could it be that the account you created the PAT for has some permission limitations? As in, it is not an site collection admin?

**Comment by jaredfholgate** [comment]
You could also attempt to create it directly via the API to see if you get the same error: https://learn.microsoft.com/en-us/rest/api/azure/devops/git/repositories/create?view=azure-devops-rest-7.1&tabs=HTTP#create-a-repository

E.g.

```pwsh
$token = [System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes(":$pat"))
$headers = @{
    Authorization = "Basic $token" 
}

$uri = "https://dev.azure.com/$org/_apis/projects/$project"

$response = Invoke-RestMethod -Uri $uri -Method Get -ContentType "application/json" -Headers $headers

$projectId = $response.id

$body = @{
    name = "testing123"
    project = @{
        "id" = $projectId
    }
}

$uri = "https://dev.azure.com/$org/_apis/git/repositories?api-version=7.1"

$response = Invoke-RestMethod -Uri $uri -Method Post -ContentType "application/json" -Headers $headers -Body ($body | ConvertTo-Json)

Write-Host "Repository created with ID: $($response.id)"
```

**Comment by DylanRobinson** [comment]
@jaredfholgate Thank you for that example - I was able to replicate the issue using that which made it much faster to debug.
I found the culprit to be **GitHub Advanced Security** which we have set to 'Automatically enable Advanced Security for new projects' at the DevOps organisation level.

Disabling Advanced Security for all new projects, or at minimum for new repositories within the existing Landing Zone project, allows the bootstrap to complete successfully. The broken repositories must be deleted if disabling Advanced Security at the project level before re-running the bootstrap.

It's not immediately clear to me why Advanced Security is causing the issue.

**Comment by jaredfholgate** [comment]
Thanks for the feedback, that is useful to know. I'll see if I can escalate this internally. Hopefully this issue will pop up in search results and help anyone else hitting the issue in the meantime. Given this is not something we can solve for (that I am aware of), I am going to close this issue for now.

**Comment by jaredfholgate** [comment]
I do wonder if incrementing the API version would fix it? `7.2-preview.1`

If that did fix it, it would require upstream changes to the GO SDK and Terraform Provider to resolve it.

**Comment by jaredfholgate** [comment]
I just tried reproducing this in my own org and was unable to. I wonder if there is another setting or permission impacting your org?
